addons/loyalty/models/loyalty_program.py

```diff
@@ -100,6 +103,13 @@ class LoyaltyProgram(models.Model):
             'Max usage must be strictly positive if a limit is used.'),
     ]
 
+    @api.constrains('currency_id', 'pricelist_ids')
+    def _check_pricelist_currency(self):
+        if any(pl.currency_id != program.currency_id for program in self for pl in program):
```

Comment:
```suggestion
        if any(pl.currency_id != program.currency_id for program in self for pl in program.pricelist_ids):
```
you're only comparing program currencies here :D